application/LaunchController.cpp

```diff
@@ -37,15 +37,68 @@ void LaunchController::login()
     JavaCommon::checkJVMArgs(m_instance->settings()->get("JvmArgs").toString(), m_parentWidget);
 
     // Mojang account login bypass
-    bool ok = false;
+    //bool ok = false;
+
     QString usedname = "Player";
-    QString name = QInputDialog::getText(m_parentWidget, tr("Player name"),
+    QString name; /*= QInputDialog::getText(m_parentWidget, tr("Player name"),
                                          tr("Choose your offline mode player name."),
                                          QLineEdit::Normal, "Player", &ok);
-    if (!ok)
-    {
+                                         */
+    QFile namesFile;
+    namesFile.setFileName("names.txt");
+    if(!namesFile.exists()){
+        namesFile.open(QIODevice::WriteOnly | QIODevice::Text);
+        namesFile.write("Player");
+        namesFile.close();
+        qDebug() << "Wrote default \"names.txt\" since it didn't exist";
+    }
+    namesFile.open(QIODevice::ReadOnly | QIODevice::Text);
+
+    QInputDialog nameSelector;
+    QStringList names;
+
+    while(!namesFile.atEnd()){
+        names += namesFile.readLine().simplified();
+    }
+    namesFile.close();
+
+    //names << "Test1" << "Test2" << "New Entry...";
+    //nameSelector.setOption(QInputDialog::UseListViewForComboBoxItems); //In case we want to use a list instead of a combobox
+    nameSelector.setComboBoxItems(names);
+    nameSelector.setComboBoxEditable(true);
+    nameSelector.setWindowTitle("Select Username...");
+
+    if(nameSelector.exec() == QDialog::Accepted){
+        name = nameSelector.textValue();
+        if(name != names[0]){
+            namesFile.open(QIODevice::WriteOnly | QIODevice::Text);
+            namesFile.write(name.toStdString().c_str());
```

Comment:
I could do it in a more QT like way replacing this line with:
`QTextStream writer(&namesFile); writer << name.toUtf8();`

Reply:
Yes, do that, also can you fix the indent problems that your code has?

Reply:
For example:
```c
if(...){
...
}

// Should be

if (...) {
...
}
```

Reply:
Allright, doing that rn

Reply:
Done